tests/returntypes/test_autograd_qnode_new.py

```diff
@@ -0,0 +1,1942 @@
+        assert hess.shape == (2, 2)
+
+    def test_multiple_derivative_var_multiple_params(self, dev_name, diff_method, mode):
+        """The hessian of single a measurement with multiple params return a tuple of arrays."""
+        dev = qml.device(dev_name, wires=2)
+
+        if diff_method == "adjoint":
+            pytest.skip("Test does not supports adjoint because second order diff.")
+
+        par_0 = qml.numpy.array(0.1, requires_grad=True)
+        par_1 = qml.numpy.array(0.2, requires_grad=True)
+
+        @qnode(dev, interface="autograd", diff_method=diff_method, max_diff=2)
+        def circuit(x, y):
+            qml.RX(x, wires=[0])
+            qml.RY(y, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.var(qml.PauliZ(0) @ qml.PauliX(1))
+
+        def cost(x, y):
+            return anp.hstack(qml.grad(circuit)(x, y))
+
+        hess = qml.jacobian(cost)(par_0, par_1)
+
+        assert isinstance(hess, tuple)
+        assert len(hess) == 2
+
+        assert isinstance(hess[0], np.ndarray)
+        assert hess[0].shape == (2,)
+
+        assert isinstance(hess[1], np.ndarray)
+        assert hess[1].shape == (2,)
+
+    def test_multiple_derivative_var_multiple_param_array(self, dev_name, diff_method, mode):
+        """The hessian of single measurement with a multiple params array return a single array."""
+        if diff_method == "adjoint":
+            pytest.skip("Test does not supports adjoint because second order diff.")
+
+        dev = qml.device(dev_name, wires=2)
+
+        params = qml.numpy.array([0.1, 0.2], requires_grad=True)
+
+        @qnode(dev, interface="autograd", diff_method=diff_method, max_diff=2)
+        def circuit(x):
+            qml.RX(x[0], wires=[0])
+            qml.RY(x[1], wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.var(qml.PauliZ(0) @ qml.PauliX(1))
+
+        hess = qml.jacobian(qml.grad(circuit))(params)
+
+        assert isinstance(hess, np.ndarray)
+        assert hess.shape == (2, 2)
+
+    def test_multiple_derivative_probs_expval_multiple_params(self, dev_name, diff_method, mode):
+        """The hessian of multiple measurements with multiple params return a tuple of arrays."""
+        dev = qml.device(dev_name, wires=2)
+
+        if diff_method == "adjoint":
+            pytest.skip("Test does not supports adjoint because second order diff.")
+
+        par_0 = qml.numpy.array(0.1, requires_grad=True)
+        par_1 = qml.numpy.array(0.2, requires_grad=True)
+
+        @qnode(dev, interface="autograd", diff_method=diff_method, max_diff=2)
+        def circuit(x, y):
+            qml.RX(x, wires=[0])
+            qml.RY(y, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0) @ qml.PauliX(1)), qml.probs(wires=[0, 1])
+
+        def circuit_stack(x, y):
+            return anp.hstack(circuit(x, y))
+
+        def cost(x, y):
+            return anp.hstack(qml.jacobian(circuit_stack)(x, y))
```

Comment:
Do we need to stack here?

Reply:
Yes it is necessary because jacobian output a tuple here.